### DISCUSSION TITLE
Question about NetworkInterface resource

### DISCUSSION THREAD

**Discussion by nick-potenski** [labgrid-project/labgrid#1013] [Ideas]
Was the `NetworkInterface` resource (in `resource/base.py`) purposely never exposed or has it just not yet been needed?  Would it be reasonable to make this a usable resource instead of just a base-class for `USBNetworkInterface`?

My PC has a PCI-based NIC (instead of USB), so I am not able to use a `USBNetworkInterface` in my local configuration.  I'm definitely able & willing to do the implementation here, but I just wanted to make sure it was OK before I jumped.

**Comment by Bastian-Krause** [comment]
I have a WIP patch lying around somewhere for this. Let me check if it is complete and usable already. I'll come back to you within the next few days.

**Reply by Bastian-Krause** [reply]
Here you go: #1014. Testing much appreciated :)

**Reply by nick-potenski** [reply]
Awesome, I will give it a try. :)
